### PR TITLE
adding tests and made slight modifications to Parser for bug handling

### PR DESCRIPTION
**Pull request by matthewphua**

### CODE REVIEW

File: src/main/java/seedu/duke/Parser.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public void checkIllegalCharacter(String userInput) throws IllegalCharacterExcep
     }
 
     public void checkEmptyCommand(String parsedInput) throws InvalidCommandException {
-        if (parsedInput.isEmpty()) {
+        if (parsedInput.isBlank()) {
```
Comment (matthewphua) [inline src/main/java/seedu/duke/Parser.java:78]: this ensures that its invalid input if user just inputs whitespace for a title or other param

File: src/main/java/seedu/duke/Parser.java [hunk 2]
```diff
@@ -336,7 +336,7 @@ public boolean isFutureDate(String dateWatchedString) {
             dateFormat.setLenient(false);
             date = dateFormat.parse(dateWatchedString);
         } catch (ParseException e) {
-            Ui.print("Invalid date format. Use the following format: DD-MM-YYYY.");
+           return true; //just return true and b/c future date and invalid date are both invalid inputs.
```
Comment (matthewphua) [inline src/main/java/seedu/duke/Parser.java:339]: We already have a print statement in AddMedia I believe so this will be redundant

Comment (matthewphua) [reply src/main/java/seedu/duke/Parser.java:339]: also this will really never happen because we check for valid date before this check

**Review by naz019 — COMMENTED** [review]
Looks good. Good test coverage and renaming of constants.

**Pull request merged**